### ISSUE TITLE
PorousFlow: Stateful props + adaptivity + TET4/TRI3 elements = crash

### ISSUE THREAD

**Issue by cpgr** [idaholab/moose#14428]
## Bug Description

*Update 2* I think the stateful/adaptivity and TET4/TRI3 mesh issue is different to this, so I'll make a new issue, and keep this one for the porous flow specific crash.

~~*Update*: Looks like the combination of stateful properties, adaptivity and TET4/TRI3 element type causes an issue in general, not just in porous flow.~~

Mesh adaptivity with a tet4 mesh (in 3D) or a tri3 mesh (in 2D) causes a segfault in PorousFlow, as discovered by Daniel, see mailing list discussion https://groups.google.com/forum/#!topic/moose-users/6yUh7_WCGHA

## Steps to Reproduce

~~*Update*: change the element type to TET4 in test/tests/materials/stateful_prop/stateful_prop_adaptivity_test.i~~

The following simple input file illustrates this
```
[Mesh]
  [./mesh]
    type = GeneratedMeshGenerator
    elem_type = TRI3
    dim = 2
    # elem_type = TET4
    # dim = 3
    nx = 1
  [../]
[]

[Adaptivity]
  marker = marker
  [./Markers]
    [./marker]
      type = ValueRangeMarker
      variable = pp
      lower_bound = 0.1
      upper_bound = 1
    [../]
  [../]
[]

[GlobalParams]
  PorousFlowDictator = dictator
[]

[Variables]
  [./pp]
    initial_condition = 0
  [../]
[]

[Kernels]
  [./mass]
    type = PorousFlowMassTimeDerivative
    variable = pp
  [../]
[]

[BCs]
  [./left]
    type = PresetBC
    variable = pp
    boundary = left
    value = 1
  [../]
  [./right]
    type = PresetBC
    variable = pp
    boundary = right
    value = 0
  [../]
[]

[UserObjects]
  [./dictator]
    type = PorousFlowDictator
    porous_flow_vars = 'pp'
    number_fluid_phases = 1
    number_fluid_components = 1
  [../]
  [./pc]
    type = PorousFlowCapillaryPressureVG
    m = 0.5
    alpha = 1
  [../]
[]

[Modules]
  [./FluidProperties]
    [./simple_fluid]
      type = SimpleFluidProperties
      bulk_modulus = 1.2
      density0 = 1
      viscosity = 1
      thermal_expansion = 0
    [../]
  [../]
[]

[Materials]
  [./temperature]
    type = PorousFlowTemperature
  [../]
  [./ppss]
    type = PorousFlow1PhaseP
    porepressure = pp
    capillary_pressure = pc
  [../]
  [./massfrac]
    type = PorousFlowMassFraction
  [../]
  [./simple_fluid]
    type = PorousFlowSingleComponentFluid
    fp = simple_fluid
    phase = 0
  [../]
  [./porosity]
    type = PorousFlowPorosityConst
    porosity = 0.1
  [../]
[]

[Postprocessors]
  [./numdofs]
    type = NumDOFs
  [../]
[]

[Preconditioning]
  [./smp]
    type = SMP
    full = true
  [../]
[]

[Executioner]
  type = Transient
  end_time = 3
  dt = 1
  solve_type = Newton
  nl_abs_tol = 1e-12
[]

[Outputs]
  execute_on = 'timestep_end'
[]
```

## Impact
Can't use adaptivity with tetrahedral meshes

**Comment by WilkAndy** [comment]
What's the betting that this is due to our crazy nodal Materials not being properly copied.

**Comment by cpgr** [comment]
Odds on I would imagine, as it is happening for these elements where the number of nodes is less than the number of qps.

**Comment by WilkAndy** [comment]
Yea.... tearing hair out.

**Comment by cpgr** [comment]
It looks like this isn't just a porous flow problem. If you change the element type to `TET4` in `test/tests/materials/stateful_prop/stateful_prop_adaptivity_test.i` then it also doesn't work.

**Comment by WilkAndy** [comment]
Good to know!

**Comment by cpgr** [comment]
There is definitely an issue with stateful/adaptivity and TET4/TRI3 but it seems to be different to this so I'll make a new issue for that and keep this one for the problem identified by Daniel.

**Comment by lindsayad** [comment]
Yea this one definitely not fixed by #14167
